beacon-chain/execution/engine_client.go

```diff
@@ -555,8 +555,11 @@ func (s *Service) ReconstructFullBlock(
 	// If the payload header has a block hash of 0x0, it means we are pre-merge and should
 	// simply return the block with an empty execution payload.
 	if bytes.Equal(header.BlockHash(), params.BeaconConfig().ZeroHash[:]) {
-		payload := buildEmptyExecutionPayload()
-		return blocks.BuildSignedBeaconBlockFromExecutionPayload(blindedBlock, payload)
+		payload, err := buildEmptyExecutionPayloadForBlock(blindedBlock)
```

Comment:
The changes here and in the consensus-types package are bugfixes. Without these changes the empty payload in the genesis block causes the code to convert the capella genesis block to a bellatrix genesis block, leading to initial-sync failures (because the RPC code encodes the ssz value according to the bellatrix schema, but the context bytes tell the receiver that the block should be unmarshaled as capella).